js/adslot.js

```diff
@@ -68,90 +58,72 @@ export class AdSlot extends React.Component {
     this.unregisterSlot();
   }
 
-  getSlotId() {
-    return this.props.slotId || this.state.slotId;
-  }
+  getAdProps = () => ({
+    className: 'adBox',
+    id: this.state.slotId,
+  });
 
-  generateSlotId() {
-    return `adSlot-${dynamicAdCount++}`;
-  }
+  getSlotId = () => this.props.slotId || this.state.slotId;
 
-  mapContextToAdSlotProps() {
-    const context = this.context;
-    const mappedProps = {};
-    if (context.dfpNetworkId !== undefined) {
-      mappedProps.dfpNetworkId = context.dfpNetworkId;
-    }
-    if (context.dfpAdUnit !== undefined) {
-      mappedProps.adUnit = context.dfpAdUnit;
-    }
-    if (context.dfpSizeMapping !== undefined) {
-      mappedProps.sizeMapping = context.dfpSizeMapping;
-    }
-    if (context.dfpTargetingArguments !== undefined) {
-      mappedProps.targetingArguments = context.dfpTargetingArguments;
-    }
-    return mappedProps;
-  }
+  generateSlotId = () => `adSlot-${dynamicAdCount++}`;
 
-  doRegisterSlot() {
+  doRegisterSlot = () => {
     DFPManager.registerSlot({
-      ...this.mapContextToAdSlotProps(),
+      ...mapContextToAdSlotProps(this.context),
       ...this.props,
       ...this.state,
       slotShouldRefresh: this.slotShouldRefresh,
     });
-    if (this.props.fetchNow === true) {
+
+    if (this.props.fetchNow) {
       DFPManager.load(this.getSlotId());
     }
+
     DFPManager.attachSlotRenderEnded(this.slotRenderEnded);
   }
 
-  registerSlot() {
+  registerSlot = () => {
     if (this.state.slotId === null) {
       this.setState({
         slotId: this.generateSlotId(),
-      }, this.doRegisterSlot);
```

Comment:
been testing this on my local and i'm getting this error in my browser's console:

<img width="804" alt="screen shot 2018-01-04 at 6 52 53 pm" src="https://user-images.githubusercontent.com/433609/34585933-9cdbbbb6-f180-11e7-8709-ddaad5d8b2c9.png">

I didn't have much available time to debug this, but it seems that problem is sort of related to the this current fcn (`registerSlot()`) and `doRegisterSlot()` which is getting called when this.state.slotId is still null:

<img width="824" alt="screen shot 2018-01-04 at 7 06 53 pm" src="https://user-images.githubusercontent.com/433609/34586513-05b7ab70-f183-11e7-86c6-2025c4d003c0.png">


not certain if this is causing the error in this case, but t`his.doRegisterSlot` should passed as callback to this.setState b/c in some cases it might defer the update until later

```
https://reactjs.org/docs/react-component.html#setstate
setState() does not always immediately update the component. It may batch or defer the update until later. This makes reading this.state right after calling setState() a potential pitfall.
```

Can u please have a look at this?. thankss.

Reply:
@jaanauati for sure I will take a look at it! Thanks for your review!